includes/Core/Tracking/REST_Tracking_Consent_Controller.php

```diff
@@ -25,25 +23,13 @@
 /**
  * Class managing admin tracking.
  *
- * @since 1.0.0
+ * @since  n.e.x.t
  * @access private
  * @ignore
  */
-final class Tracking {
-
+class REST_Tracking_Consent_Controller {
```

Comment:
I removed the `register` call.

- https://github.com/google/site-kit-wp/blob/81009e1d3f1915795f1b88ac1d1c4c67198b5f2d/includes/Core/Tracking/REST_Tracking_Consent_Controller.php#L64

Since we still need to have access to the `consent` instance due we are using it inside of the REST callback.

- https://github.com/google/site-kit-wp/blob/81009e1d3f1915795f1b88ac1d1c4c67198b5f2d/includes/Core/Tracking/REST_Tracking_Consent_Controller.php#L131
- https://github.com/google/site-kit-wp/blob/81009e1d3f1915795f1b88ac1d1c4c67198b5f2d/includes/Core/Tracking/REST_Tracking_Consent_Controller.php#L109

Reply:
I would be in favor of revoking the `is_active` method instead as is the only place where is used but don't want to introduce too much refactoring into an existing class, due this goes beyond the scope of the IB.

Reply:
@mitogh, ok, then just pass the instance of the `Tracking_Consent` class created in the `Tracking` class to this class and don't create it again.

```diff
 		$this->consent                          = new Tracking_Consent( $user_options );
-		$this->rest_tracking_consent_controller = new REST_Tracking_Consent_Controller( $context, $user_options );
+		$this->rest_tracking_consent_controller = new REST_Tracking_Consent_Controller( $context, $this->consent );
```